server/src/main.rs

```diff
@@ -98,6 +105,65 @@ async fn main() -> anyhow::Result<()> {
     }
 }
 
+fn startup_sync() {
+    if !Path::new(&CONFIG.parseable.local_disk_path).exists() {
+        return;
+    }
+
+    for stream in metadata::STREAM_INFO.list_streams() {
+        let dir = StorageDir::new(&stream);
+        // if data.parquet file is not present then skip this stream
+        if !dir.parquet_path_exists() {
+            continue;
+        }
+        if let Err(e) = dir.create_temp_dir() {
```

Comment:
what is the behaviour here if the `tmp` directory exists already? I mean we need to ensure we don't overwrite, because it is highly probably that the directory exists and has some data that is still not sent to S3.

Reply:
create_dir_all  recursively builds all the missing component of path. If there is already an existing tmp, it will not overwrite anything inside